### PR TITLE
Feature 5 surveys listing

### PR DESCRIPTION
**Pull request by Jdallos**

### CODE REVIEW

**Review by EVP1989 — APPROVED** [review]
Approved with suggestion

File: src/components/screens/surveys-screen.tsx [hunk 1]
```diff
@@ -1,7 +1,153 @@
...
   return (
-    <h1>oss-management surveys screen (Not yet implemented)</h1>
-  )
-}
+    <Box>
+      <Box>
+        <Tabs value={activeTab} onChange={(_, value) => setActiveTab(value)} >
+          <Tab
+            value={SurveyScreens.ACTIVE}
+            label={`Aktiiviset kyselyt (${mockData.length})`}
```
Comment (villejuutila) [inline src/components/screens/surveys-screen.tsx:135]: Use localized strings. You can pass variables to localized strings by defining the string as e.g. `Aktiiviset kyselyt {0}` and then format it in code as `strings.formatString(strings.surveysScreen.activeSurveys, variable)`

File: src/localization/fi.json [hunk 2]
```diff
@@ -1,7 +1,8 @@
 {
   "generic": {
     "close": "Sulje",
-    "logout": "Kirjaudu ulos"
+    "logout": "Kirjaudu ulos",
+    "notImplemented": "Ei toteutettu"
```
Comment (villejuutila) [inline src/localization/fi.json:5]:
```suggestion
    "notImplemented": "Ei vielä toteutettu"
```

File: src/localization/fi.json [hunk 3]
```diff
@@ -13,5 +14,13 @@
       "url": "Osoite: {0}",
       "errorMessage": "Virheilmoitus: "
     }
+  },
+  "surveysScreen": {
+    "surveyTitle": "Kysely",
+    "screens": "Näytöt",
+    "publicationDate": "Julkaisuaika",
+    "endTime": "Päättymisaika",
+    "mostPopular": "SuosituinNäyttö",
```
Comment (villejuutila) [inline src/localization/fi.json:23]:
```suggestion
    "mostPopular": "Suosituin Näyttö",
```

File: src/components/screens/surveys-screen.tsx [hunk 4]
```diff
@@ -1,7 +1,153 @@
...
+  {
+    id: "12314",
+    kysely: "name of survey2",
+    näytöt: 4,
+    julkaisuaika: DateTime.now(),
+    päättymisaika: DateTime.now(),
+    suosituinNäyttö: "location 2",
+    vastauksia: 17
+  },
+];
```
Comment (villejuutila) [inline src/components/screens/surveys-screen.tsx:27]: Imo mock data should be moved to its own file.

File: src/components/screens/surveys-screen.tsx [hunk 5]
```diff
@@ -1,7 +1,153 @@
...
+
+/**
+ * Interface for tab panel props
+ */
+interface TabPanelProps {
```
Comment (villejuutila) [inline src/components/screens/surveys-screen.tsx:32]: Move the TabPanel to its own component.

File: src/components/screens/surveys-screen.tsx [hunk 6]
```diff
@@ -1,7 +1,153 @@
...
+  children: React.ReactNode;
+  value: SurveyScreens;
+  index: SurveyScreens;
```
Comment (villejuutila) [inline src/components/screens/surveys-screen.tsx:35]: These could be just numbers.

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: src/components/screens/surveys-screen.tsx (hunk 1 above)
Comment (Jdallos) [reply src/components/screens/surveys-screen.tsx:135]: Done

File: src/components/screens/surveys-screen.tsx (hunk 6 above)
Comment (Jdallos) [reply src/components/screens/surveys-screen.tsx:35]: I did these as numbers at first, I thought words/ enum might be clearer?

Comment (Jdallos) [reply src/components/screens/surveys-screen.tsx]: We discussed and decided to leave as is

File: src/components/surveys/TabPanel.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import { Box } from "@mui/material";
+import { SurveyScreens } from "../../types";
+
+/**
+ * Interface for tab panel props
+ */
+interface TabPanelProps {
```
Comment (villejuutila) [inline src/components/surveys/TabPanel.tsx:7]:
```suggestion
interface Props {
```

File: src/components/surveys/TabPanel.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+/**
+ * Tab Panel
+ *
+ * @param props TabPanelProps
+ * @returns Tab panel
+ */
+const TabPanel = (props: TabPanelProps) => {
+  const {children, value, index} = props;
```
Comment (villejuutila) [inline src/components/surveys/TabPanel.tsx:20]:
```suggestion
const TabPanel: React.FC<Props> = ({children, value, index}) => {
```

**Review by villejuutila — APPROVED** [review]
Approved with two comments.

**Pull request merged**